tests/test_execute_command.py

```diff
@@ -0,0 +1,41 @@
+from unittesting import DeferrableTestCase
+import sublime
+from os.path import dirname
+from LSP.plugin.core.types import ClientConfig, ClientStates, LanguageConfig
+from LSP.plugin.core.test_session import MockClient
+from LSP.plugin.core.sessions import Session
+from LSP.plugin.core.registry import windows  # , session_for_view
+from LSP.plugin.core.settings import client_configs
+
+test_language = LanguageConfig("test", ["text.plain"], ["Plain text"])
+test_commands = ["command1", "command2"]
+text_config = ClientConfig("test", [], None, languages=[test_language], commands=test_commands,)
+
+
+class LspExecuteCommandTests(DeferrableTestCase):
+
+    def setUp(self):
+        self.view = sublime.active_window().new_file()
+        self.old_configs = client_configs.all
+        client_configs.all = [text_config]
+
+    def test_execute_command(self):
+        wm = windows.lookup(self.view.window())
+        wm._configs.all.append(text_config)
+
+        session = Session(text_config, dirname(__file__), MockClient())
+        session.state = ClientStates.READY
+        wm._sessions[text_config.name] = session
+
+        self.view.run_command('lsp_execute')
+
+        # popup should be visible eventually
+        yield self.view.is_popup_visible()
+        self.view.update_popup("command1")
```

Comment:
@rwols Do you know how to select an element from the popup and hitting "enter" via the sublime api or the testing framework ?
Otherwise we can't test that the client is receiving the command as expected ?

Reply:
Good question. I don't know the answer to that. Maybe @randy3k can clarify. How do we check the content of a quick panel or select an item from the quick panel? Is this even possible?

Reply:
If what you what is just to hit the enter, you could change the content of the quick panel (every panel is a regular view) and then execute `window.run_command("select")`.

If you need more sophisticated interactions, you might need to use https://github.com/randy3k/Keypress. See [this](https://github.com/randy3k/Keypress/blob/master/tests/test.py) for an example.

Reply:
@randy3k the command (TextCommand) that I am working on shows a pannel a require user interaction, so to test it properly Iwould need to:
- check the content of the quick pannel that is generated
- select one of the elements
- press enter 

Do you know how to do that ?

Reply:
You need to get the view if of the panel, change the content of the view and execute the command `select`.